core/security_group.go

```diff
@@ -0,0 +1,173 @@
+package core
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type SecurityGroup struct {
+	ID               string   `json:"id,omitempty"`
+	Name             string   `json:"name,omitempty"`
+	AllowAllInbound  bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound bool     `json:"allow_all_outbound,omitempty"`
+	Associations     []string `json:"associations,omitempty"`
+}
+
+func (sg *SecurityGroup) Ref() SecurityGroupRef {
+	return SecurityGroupRef{ID: sg.ID}
+}
+
+// allows a reference to a security group
+type SecurityGroupRef struct {
+	ID string `json:"id,omitempty"`
+}
```

Comment:
There's no JSON marshaling tests for this struct, though I've noticed that's actually an oversight missing from nearly every single `*Ref` type we have, and needs to be addressed. But let's add it for security groups here to begin with :)